tests/python/test_function.py

```diff
@@ -0,0 +1,111 @@
+import taichi as ti
+
+
+@ti.test(experimental_real_function=True)
+def test_function_without_return():
```

Comment:
These tests are great to have! However, the problem with them is that they don't fully cover what we want to test. That is, even if  `experimental_real_function=False`, it's going to pass anyway.

We can add more tests on the C++ side and verify at the CHI IR level. Here's how it may look like:

1. Construct a CHI IR tree containing 'FuncCallStmt'
2. Run all the passes, including the inlining one.
3. Check that the IR is processed correctly (e.g. no more `FuncCallStmt`, types are correct, etc.). 
4. * Bonus point, feed this transformed IR into an actual program and see if it works as expected.